### PR TITLE
System and sysbatch jobs always have zero index

### PR DESCRIPTION
**Pull request by tgross**
System jobs do not have unique allocation Names because the index is intended to indicated the instance out of a desired count size. Because system jobs do not have an explicit count but the results are based on the targeted nodes, the index is less informative and this was intentionally omitted from the original design. (See https://github.com/hashicorp/nomad/issues/1851). This behavior has confused a lot of users and Nomad engineers, including me!

This changeset includes:
* Updating the documentation to make this existing behavior clear to users.
* Added a test condition to the scheduler tests so that the behavior is documented in code.
* Fixed a bug where `per_alloc` volumes should not have been allowed for system and sysbatch jobs.

Note that changing this behavior rather than merely documenting it is likely to be a huge lift and not practical in the short term. We'll need to backport this but I'm going to open that as a separate PR against `release/1.4.x` and backport from there because the current `per_alloc` code includes a new-in-1.5.0 feature.

### CODE REVIEW

File: scheduler/scheduler_system_test.go [hunk 1]
```diff
@@ -66,6 +66,17 @@ func TestSystemSched_JobRegister(t *testing.T) {
 	// Ensure all allocations placed
 	require.Len(t, out, 10)
 
+	// Note that all system allocations have the same name derived from Job.Name
+	allocNames := []string{}
+	for _, alloc := range out {
+		allocNames = append(allocNames, alloc.Name)
+	}
```
Comment (shoenig) [inline scheduler/scheduler_system_test.go:73]: one liner with `helper.ConvertSlice`: https://github.com/hashicorp/nomad/blob/main/helper/funcs.go#L401

**Review by shoenig — APPROVED** [review]
LGTM! just the code golf suggestion

File: scheduler/scheduler_system_test.go (hunk 1 above)
Comment (tgross) [reply scheduler/scheduler_system_test.go:73]: Cool, will do.

**Comment by tgross** [comment]
I'm going to pick up these suggestions @shoenig but I'm also realizing that the code that creates the names is _incredibly_ misleading, because it's never called for service jobs but suggests that it is. I'm going to fix this up here as well.

**Review by lgfa29 — APPROVED** [review]

**Comment by tgross** [comment]
I'm going to go ahead with merging this without touching that test code after all. This PR covers the problem with the alloc index. But when I started messing with the test code that uses a service job to test the system diffs (oops!) it breaks in weird ways that seem unrelated to this bug entirely.

**Pull request merged**